server/src/main/java/org/elasticsearch/index/mapper/ValueFetcher.java

```diff
@@ -42,5 +44,10 @@
      * @param lookup a lookup structure over the document's source.
      * @return a list a standardized field values.
      */
-    List<Object> fetchValues(SourceLookup lookup);
+    List<Object> fetchValues(SourceLookup lookup) throws IOException;
```

Comment:
I'm not sure about this idea, but it feels natural to only pass the doc ID here:

```
List<Object> fetchValues(SourceLookup lookup) throws IOException;
```

Then `SourceValueFetcher` could hang onto the `SearchLookup` and use that to access `SourceLookup` ?

Reply:
I was thinking about that too! I didn't want to do it as part if this PR because it'd make it *huge*, but I think its a good idea.

Reply:
I've pushed a refactor for this too.